src/core/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,196 @@
+import { TooltipContent } from './TooltipContent/TooltipContent';
+import { TooltipButton } from './TooltipButton/TooltipButton';
+import { forkRefs } from '../../utils/common';
+
+const baseClassName = 'fi-tooltip';
+
+export interface TooltipProps {
+  /** Tooltip content */
+  children: ReactNode;
+  /** Toggle button label for screen readers */
+  ariaToggleButtonLabelText: string;
+  /** Close button label for screen readers */
+  ariaCloseButtonLabelText: string;
+  /** Controlled tooltip open state */
+  open?: boolean;
+  /** ClassName for toggle button */
+  toggleButtonClassName?: string;
+  /** Classname for tooltip content */
+  contentClassName?: string | undefined;
+  /**
+   * Anchor element for listening to resize events.
+   * Used for triggering tooltip content arrow indicator reposition.
+   * Should preferably be the first resizable parent container, unless there is none before document body.
+   * NOTE: Does not work with inline elements as those do not support resize events.
+   */
+  anchorElement?: HTMLElement | null;
+  /** Event handler for toggle button click */
+  onToggleButtonClick?: (event: MouseEvent | KeyboardEvent) => void;
+  /** Event handler for close button click */
+  onCloseButtonClick?: (event: MouseEvent | KeyboardEvent) => void;
+}
+
+interface InnerRef {
+  forwardedRef: React.RefObject<HTMLButtonElement>;
+}
+
+type TooltipState = {
+  open: boolean;
+  contentArrowOffsetPx: number;
+  anchorElement: HTMLElement | null;
+  anchorRefObserver: ResizeObserver | null;
+};
+
+class BaseTooltip extends Component<
+  TooltipProps & InnerRef & { className?: string }
+> {
+  state: TooltipState = {
+    open: false,
+    contentArrowOffsetPx: 0,
+    anchorElement: null,
+    anchorRefObserver: null,
+  };
+
+  private toggleButtonRef: RefObject<HTMLButtonElement>;
+
+  private contentRef: RefObject<HTMLDivElement>;
+
+  constructor(props: TooltipProps & InnerRef) {
+    super(props);
+    this.toggleButtonRef = createRef();
+    this.contentRef = createRef();
+  }
+
+  private calculateContentArrowOffset = () => {
+    if (
+      !!this.toggleButtonRef &&
+      this.toggleButtonRef.current &&
+      !!this.contentRef &&
+      !!this.contentRef.current
+    ) {
+      /**
+       * Prevent content arrow posisition outside the tooltip content box
+       * The arrow is 16 px wide and has a shadow => max pos is 17px less
+       */
+      const pos =
+        this.toggleButtonRef.current.getBoundingClientRect().left -
+        this.contentRef.current.getBoundingClientRect().left;
+      const max = this.contentRef.current.getBoundingClientRect().width - 17;
+      return Math.min(pos, max);
+    }
+    return 0;
+  };
+
+  private setContentArrowOffset = () => {
+    this.setState({ contentArrowOffsetPx: this.calculateContentArrowOffset() });
+  };
+
+  componentDidMount() {
+    window.addEventListener('resize', this.setContentArrowOffset);
+    const anchorRefObserver = new ResizeObserver(() => {
```

Comment:
It seems that at least the tests don't like the ResizeObserver.

>  ReferenceError: ResizeObserver is not defined

Need to look a little how this can be fixed.

Reply:
Couldn't yet another way than mock the ResizeObserver in the tests.

Adding the following to our ` jest/jest.setup.js`:
```js
global.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
}));

```